### PR TITLE
[NPUW] Fix position_ids Shape and KV Cache Name Issue for Qwen2.5 Omni Model

### PR DESCRIPTION
**Pull request by GuoliangShiIntel**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp [hunk 1]
```diff
@@ -245,6 +245,11 @@ void ov::npuw::LLMInferRequest::infer_generate(ov::SoPtr<ov::ITensor> input_ids,
             auto prefill_out_tensor = m_prefill_request->get_tensor(m_prefill_out_ports.at(output_name));
 
             const auto& input_name = std::regex_replace(output_name, std::regex("present"), "past_key_values");
+            if (m_kvcache_in_ports.find(input_name) == m_kvcache_in_ports.end()) {
+                LOG_DEBUG("Input name " << input_name << " does not need kv cache. Skipping.");
+                continue;
+            }
+
```
Comment (GuoliangShiIntel) [inline src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:252]: self review: Qwen2.5 Omni LLM has two output: logics and hidden_states. hidden_states is used to generate audio output and no need kv cache.
Anyway, it is always better to check the name exist in list before to use it.

**Comment by dmatveev** [comment]
@AsyaPronina please have a look on this change.

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 2]
```diff
@@ -414,7 +414,11 @@ void reshape_to_static(std::shared_ptr<ov::Model> model,
         } else if (input_name.find("attention_mask") != std::string::npos) {
             new_shape = ov::PartialShape({1, kvcache_size});
         } else if (input_name.find("position_ids") != std::string::npos) {
-            new_shape = ov::PartialShape({1, input_size});
+            const auto partial_shape_size = input.get_partial_shape().size();
+            // NB: Regular LLM uses 2D shapes, Qwen2.5 VL/Omni uses 3D shapes
+            NPUW_ASSERT(partial_shape_size == 3u || partial_shape_size == 2u);
+            new_shape =
+                partial_shape_size == 3u ? ov::PartialShape({3, 1, input_size}) : ov::PartialShape({1, input_size});
```
[the remark this answers is not in the document]
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:421]: I see it is a specific of  Qwen2.5 Omni Model, might be it worth to put also comment here

File: src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp (hunk 1 above)
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:252]: kv-cache names in inputs of generate model are expected by design now, because the model is stateless. In stateless generate model key/values set up as inputs and outputs.

Could you please elaborate why do you need to put this `continue`, what does not exactly work?

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuoliangShiIntel) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:421]: I have added comment: `// NB: Regular LLM uses 2D shapes, Qwen2.5 VL/Omni uses 3D shapes`

File: src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp (hunk 1 above)
Comment (GuoliangShiIntel) [reply src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:252]: Qwen2.5 Omni Model has two output that difference with other regular LLM
here will go through `hidden_states` output name and it is not include in `m_kvcache_in_ports`.
So if we use `m_kvcache_in_ports.at(input_name)` will result issue.
image

Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:252]: Ah, got it!
Yes, right!
Could you please also put a `//FIXME: Find only matching by names outputs and copy them, having previously checked that such inputs exist` to the loop, so I can do some common fix later? Also, could I please ask you to rename from `doesn't need kv cache` to `doesn't contain kv cache`, please?

Comment (GuoliangShiIntel) [reply src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:253]: Done

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuoliangShiIntel) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:423]: Done

**Comment by AsyaPronina** [comment]
build_jenkins

**Comment by mlukasze** [comment]
build_jenkins

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmatveev) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp:423]: Sorry, but a comment is not enough here, as we default this 0th dimension to 3 for every 3-dim case now. It must be an assert, I believe

**Review by dmatveev — APPROVED** [review]
This PR multiplies our technical debt by a factor of 2 or maybe even 3.

@AsyaPronina please have a design review with the team on this use-case and how we're going to support it going further.

**Comment by AsyaPronina** [comment]
build_jenkins
